File: LittleProjects/Scala/ScalatraDoobieCrud/scalatradoobiecrud/src/main/scala/com/github/gr1f0n6x/scalatra/doobie/crud/repository/Repository.scala
package com.github.gr1f0n6x.scalatra.doobie.crud.repository

import cats.effect.IO

trait Repository[A, ID] {
  def findOne(id: ID): IO[A]

  def findAll(): IO[Seq[A]]

  def save(a: A): IO[_]

  def update(a: A): IO[_]

  def deleteOne(id: ID): IO[_]

  def deleteAll(): IO[_]
}

File: LittleProjects/Java/GuiceRedisCrud/src/main/java/guice/redis/crud/repository/Repository.java
package guice.redis.crud.repository;

import java.util.Optional;

public interface Repository<T, ID> {
    Optional<T> findOne(ID id) throws Exception;

    T save(T t) throws Exception;

    T update(T t) throws Exception;

    void deleteOne(ID id) throws Exception;
}
